src/libs/SearchUtils.ts

```diff
@@ -12,8 +13,14 @@ function getShouldShowMerchant(data: OnyxTypes.SearchResults['data']): boolean {
     });
 }
 
+function getShouldShowColumn(data: OnyxTypes.SearchResults['data'], columnName: ValueOf<typeof CONST.SEARCH_TABLE_OPTIONAL_COLUMNS>) {
+    return Object.values(data).some((item) => !!item[columnName]);
+}
+
 function getTransactionsSections(data: OnyxTypes.SearchResults['data']): SearchTransaction[] {
     const shouldShowMerchant = getShouldShowMerchant(data);
+    const shouldShowCategory = getShouldShowColumn(data, CONST.SEARCH_TABLE_OPTIONAL_COLUMNS.CATEGORY);
```

Comment:
why not simply call this in the `TransactionItem` component itself? I don't see any benefit of this being called here.